snap/info.go

```diff
@@ -1585,6 +1585,32 @@ func ReadCurrentInfo(snapName string) (*Info, error) {
 	return ReadInfo(snapName, &SideInfo{Revision: revision})
 }
 
+func ReadCurrentComponentInfo(component string, info *Info, container func(string) (Container, error)) (*ComponentInfo, error) {
+	link := filepath.Join(ComponentsBaseDir(info.InstanceName()), info.Revision.String(), component)
+
+	linkSource, err := os.Readlink(link)
+	if err != nil {
+		return nil, fmt.Errorf("cannot find current component %q for snap %q", component, info.InstanceName())
+	}
+
+	rev := filepath.Base(linkSource)
+
+	revision, err := ParseRevision(rev)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read revision %s: %s", rev, err)
```

Comment:
s/read/parse/ ? also this error message doesn't seem to contain a lot of context

Reply:
Addressed in 4b1b48ad08585acfa8a63b3c0bb61aa676eeaf90.